openedx/core/djangoapps/catalog/utils.py

```diff
@@ -136,6 +138,9 @@ def get_programs_with_type(types=None):
             if types and program['type'] not in types:
                 continue
 
+            if program.get('hidden', False) and not include_hidden:
```

Comment:
If filtering needs on this end become any more complex, we should be thinking about caching separate sets of filtered UUIDs so we're not duplicating functionality provided by the API.